code/modules/reagents/oldchem/reagents/_reagent_base.dm

```diff
@@ -21,57 +21,42 @@
 	var/process_flags = ORGANIC
 	var/admin_only = 0
 
-/datum/reagent/proc/reaction_mob(mob/M, method=TOUCH, volume) //Some reagents transfer on touch, others don't; dependent on if they penetrate the skin or not.
-	if(!istype(M, /mob/living))
-		return 0
-	var/datum/reagent/self = src
-	src = null
-
-	if(self.holder)		//for catching rare runtimes
-		if(method == TOUCH && self.penetrates_skin)
-			if(isliving(M))
-				var/mob/living/L = M
-				var/block  = L.get_permeability_protection()
-				var/amount = round(self.volume * (1.0 - block), 0.1)
-				if(L.reagents)
-					if(amount >= 1)
-						L.reagents.add_reagent(self.id,amount)
-
+/datum/reagent/proc/reaction_mob(mob/living/M, method=TOUCH, volume) //Some reagents transfer on touch, others don't; dependent on if they penetrate the skin or not.
+	if(holder)  //for catching rare runtimes
+		if(method == TOUCH && penetrates_skin)
+			var/block  = M.get_permeability_protection()
+			var/amount = round(volume * (1.0 - block), 0.1)
+			if(M.reagents)
+				if(amount >= 1)
+					M.reagents.add_reagent(id, amount)
 
 		if(method == INGEST) //Yes, even Xenos can get addicted to drugs.
-			var/can_become_addicted = M.reagents.reaction_check(M, self)
+			var/can_become_addicted = M.reagents.reaction_check(M, src)
 
 			if(can_become_addicted)
-				if(prob(self.addiction_chance) && !is_type_in_list(self, M.reagents.addiction_list))
+				if(prob(addiction_chance) && !is_type_in_list(src, M.reagents.addiction_list))
 					to_chat(M, "<span class='danger'>You suddenly feel invigorated and guilty...</span>")
-					var/datum/reagent/new_reagent = new self.type()
+					var/datum/reagent/new_reagent = new type()
 					new_reagent.last_addiction_dose = world.timeofday
 					M.reagents.addiction_list.Add(new_reagent)
-				else if(is_type_in_list(self, M.reagents.addiction_list))
+				else if(is_type_in_list(src, M.reagents.addiction_list))
 					to_chat(M, "<span class='notice'>You feel slightly better, but for how long?</span>")
 					for(var/A in M.reagents.addiction_list)
 						var/datum/reagent/AD = A
-						if(AD && istype(AD, self))
+						if(AD && istype(AD, src))
 							AD.last_addiction_dose = world.timeofday
 							AD.addiction_stage = 1
 		return 1
 
-/datum/reagent/proc/reaction_obj(obj/O, volume) //By default we transfer a small part of the reagent to the object
-	src = null						//if it can hold reagents. nope!
-	//if(O.reagents)
-	//	O.reagents.add_reagent(id,volume/3)
+/datum/reagent/proc/reaction_obj(obj/O, volume)
 	return
 
 /datum/reagent/proc/reaction_turf(turf/T, volume)
-	src = null
 	return
 
 /datum/reagent/proc/on_mob_life(mob/living/M)
-	if(!istype(M))
```

Comment:
`on_mob_life` is handled in `metabolize` and is never called directly anywhere else; `metabolize` is only ever run in the carbon `Life` loop.